srml/scored-pool/src/lib.rs

```diff
@@ -0,0 +1,373 @@
+}
+
+decl_storage! {
+	trait Store for Module<T: Trait<I>, I: Instance=DefaultInstance> as ScoredPool {
+		/// The current pool of candidates, stored as an ordered Vec
+		/// (ordered descending by score, `None` last, highest first).
+		Pool get(pool) config(): Vec<(T::AccountId, Option<T::Score>)>;
+
+		/// The current membership, stored as an ordered Vec.
+		Members get(members): Vec<T::AccountId>;
+
+		/// Size of the `Members` set.
+		MemberCount get(member_count) config(): u32;
+	}
+	add_extra_genesis {
+		config(members): Vec<T::AccountId>;
+		config(phantom): sr_std::marker::PhantomData<I>;
+		build(|
+			storage: &mut (sr_primitives::StorageOverlay, sr_primitives::ChildrenStorageOverlay),
+			config: &Self,
+		| {
+			sr_io::with_storage(storage, || {
+				let mut pool = config.pool.clone();
+
+				// reserve balance for each candidate in the pool.
+				// panicking here is ok, since this just happens one time, pre-genesis.
+				pool
+					.iter()
+					.for_each(|(who, _)| {
+						T::Currency::reserve(&who, T::CandidateDeposit::get())
+							.expect("balance too low");
+					});
+
+				/// Sorts the `Pool` by score in a descending order. Entities which
+				/// have a score of `None` are sorted to the beginning of the vec.
+				pool.sort_by_key(|(_, maybe_score)|
+					Reverse(maybe_score.unwrap_or_default())
+				);
+
+				<Pool<T, I>>::put(&pool);
+				<Module<T, I>>::refresh_members(false);
+
+				let members = <Members<T, I>>::get();
+				T::MembershipInitialized::initialize_members(&members);
+			});
+		})
+	}
+}
+
+decl_event!(
+	pub enum Event<T, I=DefaultInstance> where
+		<T as system::Trait>::AccountId,
+	{
+		/// The given member was removed. See the transaction for who.
+		MemberRemoved,
+		/// An entity has issued a candidacy. See the transaction for who.
+		CandidateAdded,
+		/// An entity withdrew candidacy. See the transaction for who.
+		CandidateWithdrew,
+		/// The candidacy was forcefully removed for an entity.
+		/// See the transaction for who.
+		CandidateKicked,
+		/// A score was attributed to the candidate.
+		/// See the transaction for who.
+		CandidateScored,
+		/// Phantom member, never used.
+		Dummy(sr_std::marker::PhantomData<(AccountId, I)>),
+	}
+);
+
+decl_module! {
+	pub struct Module<T: Trait<I>, I: Instance=DefaultInstance>
+		for enum Call
+		where origin: T::Origin
+	{
+		fn deposit_event<T, I>() = default;
+
+		/// Every `Period` blocks the `Members` set is refreshed from the
+		/// highest scoring members in the pool.
+		fn on_initialize(n: T::BlockNumber) {
+			if n % T::Period::get() == Zero::zero() {
+				<Module<T, I>>::refresh_members(true);
+			}
+		}
+
+		/// Add `origin` to the pool of candidates.
+		pub fn issue_candidacy(origin) {
+			let who = ensure_signed(origin)?;
+			ensure!(Self::find_in_pool(&who).is_err(), "already a member");
+
+			T::Currency::reserve(&who, T::CandidateDeposit::get())
+				.map_err(|_| "balance too low")?;
+
+			let mut pool = <Pool<T, I>>::get();
```

Comment:
```suggestion
			<Pool<T, I>>::mutate(|pool| pool.push((who.clone(), None)));
```

and delete next two lines.